src/iminuit/cost.py

```diff
@@ -567,11 +572,23 @@ def _prepare_data(self, value):
 
         return n
 
+    def _pred(self, args):
+        X = np.meshgrid(*self._xe) if self._ndim > 1 else self._xe
+        cdf = self._model(X, *args)
+        cdf = _normalize_model_output(cdf)
+        if self._ndim == 1:
+            d = np.diff(cdf)
+        elif self._ndim == 2:
+            d = cdf[1:, 1:] - cdf[1:, :-1] - cdf[:-1, 1:] + cdf[:-1, :-1]
+        else:
+            raise ValueError("general ND case with D > 2 currently not supported")
```

Comment:
This error could already be raised in `__init__` on checking `_ndim`, right?

Reply:
Yes, but the `if` here is comparably cheap, so it is ok. The new version does not raise anymore.